Sources/Sentry/SentryTracer.m

```diff
@@ -686,7 +698,19 @@ - (SentryTransaction *)toTransaction
 
 #if SENTRY_TARGET_PROFILING_SUPPORTED
     if (self.isProfiling) {
+        // if we have an app start span, use its app start timestamp. otherwise use the tracer's
+        // start system time as we currently do
+        SENTRY_LOG_DEBUG(@"Tracer start time: %llu", self.startSystemTime);
+
         transaction.startSystemTime = self.startSystemTime;
+#    if SENTRY_HAS_UIKIT
+        if (appStartMeasurement != nil) {
+            SENTRY_LOG_DEBUG(@"Assigning transaction start time as app start system time (%llu)",
+                appStartMeasurement.appStartSystemTimestamp);
+            transaction.startSystemTime = appStartMeasurement.appStartSystemTimestamp;
```

Comment:
`h`: The app `transaction.startSystemTime` can be quite different from the `transaction.startTimestamp` for app start transactions when the app start is not prewarmed, because then the `transaction.startTimestamp` is the process start timestamp and the `transaction.startSystemTime` then is the runtime init timestamp. I'm not sure if this is a problem.

Reply:
We don't use the NSDate-based timestamps for any alignment of transaction/span data and profile data. Also, those two timelines are unrelated, as we've discussed previously in https://github.com/getsentry/sentry-cocoa/issues/2155

Reply:
Just wanted to double-check 👍